### ISSUE TITLE
StripeObjectEncoder encodes a string, not an object

### ISSUE THREAD

**Issue by tghw** [stripe/stripe-python#18]
Currently, the `default()` method of `StripeObjectEncoder` returns a string when the object is a StripeObject (see [__init__.py:572](https://github.com/stripe/stripe-python/blob/master/stripe/__init__.py#L572)). Per the [json documentation](http://docs.python.org/library/json.html#basic-usage): "default(obj) is a function that should return a serializable version of obj or raise TypeError. The default simply raises TypeError."

In other words, `StripeObjectEncoder.default()` should only return `obj.to_dict()`.
